src/future.ml

```diff
@@ -177,6 +177,13 @@ and opened_file_desc =
   | Fd      of Unix.file_descr
   | Channel of out_channel
 
+type purpose =
+  | Internal_job
+  | Build_job of
+      Path.t list (* targets *)
+    * Path.t      (* build directory *)
+    * string      (* context name *)
```

Comment:
Sure. It was originally an inline record, until I realised that it had to build with 4.02.3